src/jit/codegenxarch.cpp

```diff
@@ -1280,42 +1280,31 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
             gcInfo.gcMarkRegSetNpt(RBM_RDX);
         }
 
-        if (divisor->isContainedIntOrIImmed())
-        {
-            GenTreeIntConCommon* divImm = divisor->AsIntConCommon();
-            assert(divImm->IsIntCnsFitsInI32());
-            ssize_t imm = divImm->IconValue();
-            assert(isPow2(abs(imm)));
-            genCodeForPow2Div(treeNode->AsOp());
-        }
+        // Perform the 'targetType' (64-bit or 32-bit) divide instruction
+        instruction ins;
+        if (oper == GT_UMOD || oper == GT_UDIV)
+            ins = INS_div;
         else
-        {
-            // Perform the 'targetType' (64-bit or 32-bit) divide instruction
-            instruction ins;
-            if (oper == GT_UMOD || oper == GT_UDIV)
-                ins = INS_div;
-            else
-                ins = INS_idiv;
+            ins = INS_idiv;
             
-            emit->emitInsBinary(ins, size, treeNode, divisor);
+        emit->emitInsBinary(ins, size, treeNode, divisor);
             
-            // Signed divide RDX:RAX by r/m64, with result
-            //    stored in RAX := Quotient, RDX := Remainder.
-            // Move the result to the desired register, if necessary
-            if (oper == GT_DIV || oper == GT_UDIV)
+        // Signed divide RDX:RAX by r/m64, with result
```

Comment:
I can update the comment but this isn't code that I have changed. The GitHub diff viewer shows changes here because it lacks a "ignore whitespace changes" option.